File: deleteDuplicates/fastestSolution.cpp
// link: https://leetcode.com/problems/remove-duplicates-from-sorted-list-ii/

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        while(head && head->next && head->val == head->next->val){
            int val = head->val;
            while(head && head->val == val){
                ListNode* t = head;
                head = head->next;
                delete t;
            }
        }
        ListNode* t = nullptr, *l = nullptr;
        if(head) l = head, t = head->next;
        while(t){
            if(t && t->next && t->val == t->next->val){
                int val = t->val;
                while(t && t->val == val){
                    ListNode* tmp = t;
                    t = t->next;
                    delete tmp;
                }
                l->next = t;
                
            }
            else {
                l = t;
                t = t->next;
            }
        }
        
        
        return head;
    }
};

File: swapNodesInPairs/fastestSolution.py
"""
Definition for singly-linked list.
class ListNode:
	def ___init___(self, val=0, next=None):
		self.val = val
		self.next = next
"""
class Solution:
	def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
		if head and head.next:
			ret, ptr, pre = head.next, head, None

			while ptr:
				if not ptr.next:
					break
				ptr.next, ptr, ptr.next = ptr.next.next, ptr.next, ptr

				if pre:
					pre.next = ptr

				pre, ptr = ptr.next, ptr.next.next
			return ret
		else:
			return head